### PR TITLE
Cache Cody in CI

### PR DESCRIPTION
**Pull request by mkondratek**
With cache:  [5m 24s](https://github.com/sourcegraph/jetbrains/actions/runs/9408161568)
Without cache:   [6m 14s](https://github.com/sourcegraph/jetbrains/actions/runs/9384842830)

## Test plan
1. CI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,14 +19,16 @@ jobs:
         uses: actions/cache@v3
         with:
           path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
-          key: ${{ runner.os }}-${{ matrix.node }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
```
Comment (mkondratek) [inline .github/workflows/test.yml:22]: There is no `pnpm-lock.yaml` in our repo. `hashFiles('**/pnpm-lock.yaml')` is always `""` (empty string)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -19,14 +19,16 @@ jobs:
...
-      - uses: actions/cache@v3
+      - name: Cache gradle and sourcegraph stores
+        uses: actions/cache@v3
         with:
           path: |
             ~/.gradle/caches
             ~/.gradle/wrapper
-          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+            ~/.sourcegraph/caches
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties', 'gradle.properties') }}
```
Comment (mkondratek) [inline .github/workflows/test.yml:31]: `gradle.properties` changes with the cody commit bumps

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -19,14 +19,16 @@ jobs:
...
           path: ${{ steps.pnpm-cache.outputs.STORE_PATH }}
-          key: ${{ runner.os }}-${{ matrix.node }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          key: ${{ runner.os }}-${{ matrix.node }}-pnpm-store-
           restore-keys: ${{ runner.os }}-${{ matrix.node }}-pnpm-store-k
-      - uses: actions/cache@v3
+      - name: Cache gradle and sourcegraph stores
+        uses: actions/cache@v3
         with:
           path: |
             ~/.gradle/caches
```
Comment (mkondratek) [inline .github/workflows/test.yml:28]: I wonder - is it ok to cache IntelliJ artefacts? They seem to take a lot of space 

This is an example of a repo that removes them 😄 
https://github.com/VirtusLab/git-machete-intellij-plugin/blob/8910e34c833779b5f8261fca298aeb549b80c875/.circleci/config.yml#L198-208

Comment (mkondratek) [reply .github/workflows/test.yml:28]: The cache size limit is 10 GB.

File: src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt [hunk 4]
```diff
@@ -6,58 +6,78 @@ class PositionTest : BasePlatformTestCase() {
...
     myFixture.openFileInEditor(file)
-  }
 
-  fun test_isStartOrEndOfDocumentMarkerReturnsTrueWhenLineIsLessThanZero() {
     val position = Position(-1, 0)
     val result = position.isStartOrEndOfDocumentMarker(document)
     assertEquals(true, result)
   }
 
   fun test_isStartOrEndOfDocumentMarkerReturnsTrueWhenLineIsGreaterThanLineCount() {
```
Comment (pkukielka) [inline src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt]: Is that accidentally there?

Comment (mkondratek) [reply src/test/kotlin/com/sourcegraph/cody/agent/protocol/PositionTest.kt]: it was rebased on the other branch with failing test fix/investigations - rebased back to main

**Review by pkukielka — APPROVED** [review]
LGTM

**Pull request merged**